pkg/test/xds/builders/proxy_builder.go

```diff
@@ -64,3 +64,8 @@ func (p *ProxyBuilder) WithRouting(routing xds.Routing) *ProxyBuilder {
 	p.res.Routing = routing
 	return p
 }
+
+func (p *ProxyBuilder) WithRoutingBuilder(routing *RoutingBuilder) *ProxyBuilder {
+	p.res.Routing = *routing.Build()
+	return p
+}
```

Comment:
Nit: In other places you don't have methods for both `Foo` and `FooBuilder` and instead `Foo` takes `FooBuilder` as an argument instead of `Foo`. I would try to be consistent here.

Reply:
I forgot to changer it, thank you for pointing that out